modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthAdminServiceTestCase.java

```diff
@@ -81,9 +82,48 @@ public void testRegisterApplication() throws Exception {
 
     }
 
+    @Test(groups = "wso2.is", description = "Testing retrieving Oauth2 application state", dependsOnMethods = "testRegisterApplication")
+    public void testGetOauthAppState() throws Exception {
+
+        try {
+            getOauthApplicationState();
+        } catch (Exception e) {
+            Assert.assertFalse(false, "Error when retrieving oauth app state");
+        }
+    }
+
+    @Test(groups = "wso2.is", description = "Testing updating Oauth2 application state", dependsOnMethods = "testRegisterApplication")
+    public void testUpdateOauthAppState() throws Exception {
+
+        boolean updated = false;
+
+        try {
+            updateConsumerAppState(OAuth2Constant.OAUTH_APPLICATION_STATE_REVOKED);
+            if(getOauthApplicationState().equalsIgnoreCase(OAuth2Constant.OAUTH_APPLICATION_STATE_REVOKED)) {
+                updated = true;
+            }
+        } catch (Exception e) {
+            Assert.assertFalse(false, "Error when retrieving oauth app state");
+        }
+        Assert.assertTrue(updated);
+    }
+
+    @Test(groups = "wso2.is", description = "Testing updating Oauth2 application state", dependsOnMethods = "testRegisterApplication")
+    public void testUpdateOauthSecretKey() throws Exception {
```

Comment:
it's not depends on testGetOauthAppState, it depends on testRegisterApplication.